torch/distributed/distributed_c10d.py

```diff
@@ -750,9 +745,10 @@ def _check_valid_timeout(timeout: Any) -> None:
 STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
 
 
-def _get_pg_default_device(group: Optional[ProcessGroup] = None) -> torch.device:
+def _get_object_coll_device(group: Optional[ProcessGroup] = None) -> str:
```

Comment:
can you say more about what the users are doing with this API outside of this file?  

1) renaming it won't make people stop using the new name
2) its an `_` api anyway, so we are free to change it i guess?

Reply:
Here is a search of references:
https://github.com/search?q=repo%3Apytorch%2Fpytorch%20_get_pg_default_device&type=code
Mainly in DCP.

Reply:
Yeah, renaming or `_` won't stop people from using it. I just want to make the name clear so users know what it does.

Reply:
tbh i feel like 'default device' was more clear and i dont really know what 'object_coll_device' means at all.  (unless i read the code for object coll, realize that it needs to infer a device without a tensor to look at).

Do you think it makes sense to just deprecate the fn without renaming it?  idk..

Reply:
Sorry, I am a bit confused. I am deprecating `_get_pg_default_device`. But I also need a function to hold the logic for inferring a device for object coll. 

The `dont really know what 'object_coll_device' means at all` seems like a bonus because then people won't easily use it outside.

Reply:
> Here is a search of references:
https://github.com/search?q=repo%3Apytorch%2Fpytorch%20_get_pg_default_device&type=code
Mainly in DCP.

It looks like DCP and FSDP and mostly using similar to the `_device_capability()` method you added. So maybe that should be updated?

Reply:
Yeah, thanks for mentioning that. I was in side chat with @LucasLLC @wz337 @fegin earlier. Maybe we should file an issue to migrate the usage?